src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
```

Comment:
I would argue that having a string to be returned as a date feels more natural and reuses existing parameters, also as per @nicolas-grekas suggestion.

Take the following examples. I would argue that the second one is more verbose and redundant as the timezone needs to be specified twice, which is why I ended up with a string as the default (in the original PR the default parameter was actually a datetime object as you suggest).

```
 $fromDate = $bag->getDate('fromDate', 'Y-m-d', '2016-01-01', new \DateTimeZone('CET'));
```


```
$timezone = new \DateTimeZone('CET')
$default = new \DateTime('2016-01-01', $timezone);
$fromDate = $bag->getDate('fromDate', 'Y-m-d',  $default, $timezone);
```